MAINTAINERS: initial version of MAINTAINERS file

ioannisg opened the pull request:
Pushing the initial version of Zephyr MAINTAINERS file.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Data base for holding maintainer-ship info in Zephyr

- Aligns with TSC terminology (Maintainers, Collaborators)
- Covers most of the Zephyr source Tree (initial source: https://docs.google.com/spreadsheets/d/1QxfZuM1qQIXy_EJldQeS6nErt7DjHw3SZ2vfiKVdo5M/edit#gid=0)
- Adds orphaned areas as well

- ~~Requires TSC approval~~ (motion approved)

ioannisg committed:
MAINTAINERS: initial version of MAINTAINERS file

Pushing the initial version of Zephyr MAINTAINERS file.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

stephanosio commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1146 @@
+        - samples/drivers/entropy/
+        - tests/drivers/entropy/
+
+"Drivers: Entropy: OpenISA RV32M1":
+    # Provisional entry, to be corrected
+    collaborators:
+        - MaureenHelm
+    files:
+        - drivers/entropy/*rv32m1*
```
missing `status:`

stephanosio commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1146 @@
+    labels:
+        - "area: ADC"
+
+"Drivers: ADC stm32":
+    # Provisional entry, to be corrected
+    collaborators:
+        - cybertale
+    files:
+        - drivers/adc/adc_stm32.c
+        - dts/bindings/iio/adc/st*stm32-adc.yaml
```
missing `status:`

stephanosio reviewed (commented):
I suppose missing `status:` is not intentional?

nashif committed:
scripts: add maintainer script

Adding the get_maintainer.py script

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

alexanderwachter commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1136 @@
+
+"Drivers: lora":
+    # Provisional entry, to be checked
+    status: orphaned
```
I think this should be @Mani-Sadhasivam

Mani-Sadhasivam commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1132 @@
+
+"Drivers: LED":
+    # Provisional entry, to be checked
+    status: orphaned
```
I still maintain this.

Mani-Sadhasivam commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1132 @@
+
+"Drivers: lora":
+    # Provisional entry, to be checked
+    status: orphaned
```
This one too

erwango commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1136 @@
+    maintainers:
+        - galak
+    collaborators:
+        - ulfalizer
```
Is @ulfalizer still active?

ioannisg replied:
Not sure, so I removed the entry.

erwango commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1136 @@
+        - samples/drivers/crypto/
+    labels:
+        - "area: Crypto / RNG"
+
```
Miss DAC

ioannisg replied:
Added DAC, checked with @martinjaeger for availability

nashif commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1136 @@
+#     labels:
+#         List of GitHub labels to add to pull requests that modify the area.
+#
+#     files:
```
I would like to see the syntax unified here and using the same globbing we already use elsewhere in the project, having two section for files with regex support will only make it difficult to integrate with existing CI actions and workflows.

ioannisg replied:
@nashif not sure I get it 100%: 
So you propose to have a single files-regex: property, so we would only be including files coming from regular expressions?

ioannisg committed:
CODEOWNERS: auto-assign reviews in MAINTAINERS.yml, get_maintainer.py

Add an entry in CODEOWNERS for auto-assigning reviewers
in MAINTAINERS.yml and get_maintainer.py files.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
The Maintainer's database should be ready for final review:

- I have updated the database based on the outcome of the Virtual F2F, see here: https://docs.google.com/document/d/1BDLlEh5_YuSiaQURvSO7JkHoF50cy9AD1kMOeDhO1ms/edit#
- I reached out to new candidate maintainers for availability, but added everyone also in review-here so they can +1 and confirm acceptance of the new role
- TSC only needs to approve _changes_ to maintenance status, marked in the database with 
`    # Provisional entry, subject to approval`
TSC only needs to approve maintainers (not added collaborators)

ioannisg commented:
FYI: @ruuddw, @trond-snekvik @jhedberg @cvinayak @tejlmand @andrewboie @nashif @nvlsianpu @galak @vanwinkeljan @carlescufi @anangl @Vudentz @alexanderwachter @nordic-krch @ceolin @martinjaeger @henrikbrixandersen @tbursztyka @mnkp @microbuilder @pabigot @Mani-Sadhasivam @mbolivar-nordic @mike-scott @dcpleung @jukkar @MaureenHelm @loicpoulain @de-nordic @aescolar @pfalcon @jfischer-phytec-iot @wentongwu @nategraff-sifive @jakub-uC @erwango

trond-snekvik reviewed (approved):
Approving entry for Bluetooth Mesh

pfalcon commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1166 @@
+        - "area: Clock control"
+
+"Drivers: Console":
+    status: orphaned
```
This is not correct entry. There should be 2 entries: "Subsys: Console", with include/console/ and subsys/console/, and myself as a maintainer. And "Drivers: Console" with include/drivers/console/ and drivers/console/. That's part which have wonders like handling line editing in ISR. Glad to hear it's orphaned, should be gone altogether. Feel free to keep me in "collaborators", so somebody watched what happens there.

ioannisg replied:
Sure @pfalcon i'll add this entries as you propose (will need to go via TSC approval)

alexanderwachter commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1166 @@
+        - boards/arm/nucleo_*/
+        - boards/arm/stm32*_disco/
+        - boards/arm/stm32*_eval/
+        - drivers/*/*stm32*/
```
Don‘t we create lots of conflicts with this entry?

erwango replied:
I'm happy to remove if this is not in line with general agreement and we expect there's only one assignee per PR.

ioannisg replied:
Yeap, I don't see a problem, I think @alexanderwachter might be referring to conflict for assignee-picking not conflicts in reviews.
when it comes to file/directory coverage, we do have conflicts allover the place, we just need to use some smart ways to pick assignees in PRs.

In this particular example CAN area and STM32 area both have maintainers appointed; for a PR changing an STM32 CAN drivers any of the respective maintainers may be assignees. We should be using some common sense here.
Otherwise, if a vendor area does not have an explicit maintainer, then Driver-Area maintainers become assignees for such PRs.

alexanderwachter replied:
It is just about the script. It would be annoying if the assignee picking is not working correctly. If we could have the maintainer as assignee and @erwango as reviewer, that would be perfect. But before such mechanism is working, we should be careful.

ioannisg replied:
Yeap, we should take care of this, thanks @alexanderwachter

tbursztyka commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1166 @@
+        - "area: Pinmux"
+
+"Drivers: PTP Clock":
+    status: orphaned
```
jukkar could be set as maintainers there

ioannisg replied:
Agreed, I already tried to reach him, to confirm he is available, but seems he is off-work. Not sure I want us to appoint him without him accepting. Can always be done at a later point.

ioannisg replied:
Updated this now @jukkar @tbursztyka

tbursztyka commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1166 @@
+"Networking: Buffers":
+    status: maintained
+    maintainers:
+        - tbursztyka
```
switch maintainers/collaborators: jhe is the maintainer of this part.

ioannisg replied:
That's ok.
@jhedberg ^^

jhedberg replied:
That's fine with me. I've always viewed it as us having shared maintainership, but if you want to put just one name I don't object to mine being there.

ioannisg replied:
sure, done now

henrikbrixandersen reviewed (approved):
Approved my EEPROM and CAN entries.

avisconti commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1166 @@
+"Drivers: Sensors":
+    status: maintained
+    maintainers:
+        - MaureenHelm
```
collaborators:
        - avisconti

avisconti replied:
I discussed some time ago with @MaureenHelm 
I can be a collaborator here.

ioannisg replied:
sure, i'll update this already.

ioannisg replied:
done now, @avisconti

avisconti replied:
@ioannisg 
Thanks

erwango reviewed (commented):
- Should we keep "labels" keyword ? (since it is now covered by GH action)
- s/maintainers/maintainer
- Don't we miss a dts area?

ioannisg commented:
> * Should we keep "labels" keyword ? (since it is now covered by GH action)

Good point, I'd like @nashif to comment here.

> * s/maintainers/maintainer

This was discussed in the F2F. Although we agreed that it's probably good to have a single maintainer, we might want to have multiple ones for one or several areas, so i think it is better to keep "maintainers" at the moment for backwards-compatibility. We won't have to change the grammar later, if we decide differently. Also matches with "collaborators" which is also in plural.

> * Don't we miss a dts area?
We do have a "Device Tree" area, not sure what you refer to.

Mani-Sadhasivam reviewed (approved):
LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.

pabigot commented:
I'm willing to be maintainer for I2C.

dcpleung reviewed (approved):
Approved for UART drivers, PCIe drivers, and Xtensa arch.

andysan commented:
> LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.

Happy to help out with LoRa and LoRaWAN if needed.

martinjaeger reviewed (approved):
DAC driver entry approved from my side.

MaureenHelm reviewed (approved):
Thanks @ioannisg !

mike-scott commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1178 @@
+
+"Networking: LWM2M":
+    status: maintained
+    maintainers:
```
@ioannisg LwM2M subsystem now maintained by @rlubos

ioannisg replied:
Great, thanks, I updated the entry. Will need, formally, TSC approval.

nashif commented:
> Good point, I'd like @nashif to comment here.

ideally we should use just this file for all this information and drop the file dedicated to labels.

anangl reviewed (approved):
Approved the ADC, I2S, and PWM entries.

de-nordic commented:
+1 On mcumgr and VFS entries. It might be worth asking @nvlsianpu  and @pabigot  to be collaborators on VFS.

ioannisg commented:
> I'm willing to be maintainer for I2C.

Thanks @pabigot , updated the entry, pls, +1.

ioannisg commented:
> > LED and LoRa part LGTM. I'd like @andysan to be a collaborator if he is interested but that can be done later also.
> 
> Happy to help out with LoRa and LoRaWAN if needed.

Collabs can be added at later stage

erwango commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1184 @@
+"Drivers: PWM":
+    # Provisional entry, subject to approval
+    status: maintained
+    maintainers: anangl
```
Should it be rather
```suggestion
    maintainers: 
        - anangl
```
?

ioannisg replied:
Yes, thanks. Fixed now. Also similar occurrences of this problem.

jukkar commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1188 @@
+    status: maintained
+    maintainers:
+        - pfalcon
+    files:
```
Could I be added as a collaborator here?

jukkar commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1188 @@
+    maintainers:
+        - jhedberg
+    collaborators:
+        - tbursztyka
```
and also here

jukkar commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1188 @@
+"Networking: CoAP":
+    status: maintained
+    maintainers:
+        - rveerama1
```
This needs to change. @rlubos could you take maintainership of the CoAP stuff?
I could be added as a collaborator here too.

rlubos replied:
@jukkar I think I can take it as well since it's tightly related to LwM2M anyway.
This might need TSC approval, right @ioannisg?

ioannisg replied:
Yeap, this is right. Let me change this then.

jukkar commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1188 @@
+    status: maintained
+    maintainers:
+        - rlubos
+    files:
```
Could I be added as a collaborator here.

pabigot reviewed (approved):
accept maintainer for I2C and littlefs

ioannisg commented:
@jukkar it's not needed to add now collaborators, can be done after we merge the initial version, without need for TSC approval

wentongwu commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1193 @@
+    # Provisional entry, subject to approval
+    status: maintained
+    maintainers:
+        - wentongwu
```
I think we need add @pabigot here.

ioannisg replied:
Yeap, we could do this later, though.

ioannisg commented:
> > Good point, I'd like @nashif to comment here.
> 
> ideally we should use just this file for all this information and drop the file dedicated to labels.

Well, this should be the case, at least for the "Area: xxx" labels, which map to code maintenance areas. 
For other types of labels, @nashif , I am not sure if we want to use this file.

nashif commented:
> Well, this should be the case, at least for the "Area: xxx" labels, which map to code maintenance areas.
> For other types of labels, @nashif , I am not sure if we want to use this file.

why do we want to do labels in 2 different places and have paths done twice? This will just lead to conflicting data. But regardless of that, I still think we should use the same format for paths that is not spread between different keys the same way we do it in the auto-labeler to avoid having to deal with 2 different formats that needs to be supported in CI.

The format suggested can be seen here: https://github.com/actions/labeler#usage

ioannisg commented:
> why do we want to do labels in 2 different places and have paths done twice? This will just lead to conflicting data.

I don't know how much of a problem this is. I am only saying that it makes sense for the Maintainers' database to do labels for "areas", but we might want also to be adding different kind of labels (i.e. not related to Areas) and _that_ set of labels should probably not be listed in the MAINTAINERS.yml

I don't think this could we conflicting: we should just do "union" of whatever is generated by each script.

nashif commented:
> I don't think this could we conflicting: we should just do "union" of whatever is generated by each script.

the issue now is about the files format, lets put labeling aside. We agreed to merge this as is and do the change to file format later, so I am fine with this.

abrodkin commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1193 @@
+    maintainers:
+        - ruuddw
+    collaborators:
+        - vonhust
```
Again, @vonhust is leaving so I would suggest to add me @abrodkin,  Watson (@IRISZZW) & Eugeniy (@Palmyr3) here in collaborators.

ioannisg replied:
Sure, I 'll make this change, since this is only about collaborators.

ioannisg replied:
@evgeniy-paltsev, I suppose.

abrodkin replied:
> 
> 
> @evgeniy-paltsev, I suppose.

Right, he just switched his GitHub account to be easier matched with real name :)

ioannisg replied:
done now @abrodkin

abrodkin commented on CODEOWNERS:
```diff
@@ -17,6 +17,7 @@
 /.github/workflows/                       @galak @nashif
 /.buildkite/				  @galak
+/MAINTAINERS.yml                          @ioannisg @MaureenHelm
 /arch/arc/                                @vonhust @ruuddw
```
@vonhust is leaving Synopsys, so pls mention @abrodkin instead.

carlescufi replied:
This is done now @abrodkin

abrodkin reviewed (changes requested):
@vonhust is leaving Synopsys so it's good to specify other people instead of him for ARC stuff.

ioannisg committed:
CODEOWNERS: replace @vonhust with @abrodkin for ARC review requests

Add @abrodkin to the list of CODEOWNERS for ARC code (ARCH,
BOARDS, DTS) instead of @vonhust.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

abrodkin reviewed (approved):
LGTM now, thanks

bwasim commented on MAINTAINERS.yml:
```diff
@@ -0,0 +1,1198 @@
+    labels:
+        - "area: PCI"
+
+"Drivers: peci":
```
Typo, Should be "pcie".. Same below..

stephanosio replied:
This is not a typo.
